### ISSUE TITLE
Feature Request: Add Support for Maiden Names

### ISSUE THREAD

**Issue by tbirrell** [monicahq/monica#2152]
This is something I've added to my personal instance. But I think it would be good to add to the main project. At the moment I just have an extra colum in my DB that I edit from my IDE. But from the front, it looks something like this:

![image](https://user-images.githubusercontent.com/17275888/49890925-e894e400-fe0a-11e8-8a85-b87ce13549eb.png)

If whoever works on this (assuming its not me) wants my code to use as a foundation, I have a migration for the DB, some extra methods in `Contact.php`, and the change to `_header.blade.php` (which is visible in the above picture). What I don't have is any sort of form for adding/editing the information, or any code in the Controller to save it.

**Comment by mccollam** [comment]
Does it make sense to generalize this some?  People change names for several reasons, sometimes first name as well.  I've been using the description field to capture this now, but it's free-form (e.g. "birth name was XXX").

This may be too specific to my contacts but I could see a use for tracking name changes and then allowing searches even on old names.

**Comment by tbirrell** [comment]
@mccollam That would probably require a significant redesign of `monica.contacts`. Possibly even moving names completely out to a pivot table. This is enough of a change to justify its own feature request ticket, I think.

**Comment by asbiin** [comment]
I disagree with the term "maiden name" which is associated only to women, and thus could be taken as a sexist term. I prefer something ile "birth name".

**Comment by mccollam** [comment]
Actually "birth name" would capture a lot of my cases as well (someone changes their name, changes gender, etc.)

**Comment by tbirrell** [comment]
> I disagree with the term "maiden name" which is associated only to women, and thus could be taken as a sexist term. I prefer something ile "birth name".

It's a legal term though. So should not be a problem. It is also very specifically for recording pre-marriage names, as you can see from how I display it on the frontend. 

If we want to put this on hold until we can come up with a more comprehensive solution for alternative names, I'm cool with that.

**Comment by trwnh** [comment]
"Former name" would cover more use cases than "birth name". As an edge case, consider a case where a person has a legal name change, twice.

**Comment by Alwaysin** [comment]
Was searching for where to input maiden names, didn't think it wouldn't be there :(
I put the info in the description of the contact for the time being.

**Comment by immanuelfodor** [comment]
I've just run into this issue as I was looking for a way to store former name for a person who changed name after marriage. I also have some people who legally changed names even as male because they didn't like their given name. Former name +1
